flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamTaskStateInitializerImpl.java

```diff
@@ -266,6 +266,11 @@ protected OperatorStateBackend operatorStateBackend(
 			taskInfo.getNumberOfParallelSubtasks(),
 			taskInfo.getIndexOfThisSubtask());
 
+		// Now restore processing is included in backend building/constructing process, so we need to make sure
+		// each stream constructed in restore could also be closed in case of task cancel, for example the data
+		// input stream opened for serDe during restore.
+		CloseableRegistry cancelStreamRegistry = new CloseableRegistry();
+		backendCloseableRegistry.registerCloseable(cancelStreamRegistry);
```

Comment:
This registry is never unregistered and can be closed twice now, once by itself and as we pass it into the backend also through the backends `close/dispose` methods. We could either work with two registries, once scoped for the restore and once scoped for the backend object itself, or at least you can unregister the registry once the restore is done, because the backend is registered and `cancelStreamRegistry` is closed as part of the backend's `close/dipose` now.

Reply:
Will follow the former way with two registries since it shows the semantic clearly.